app/src/main/java/com/android/sample/model/recipe/networkData/MealDBRecipesRepository.kt

```diff
@@ -211,7 +212,16 @@ class MealDBRecipesRepository(private val client: OkHttpClient) : RecipeNetworkR
             })
   }
 
-  override fun listCategories(onSuccess: (List<String>) -> Unit, onFailure: (Exception) -> Unit) {
+  override fun filterSearch(
```

Comment:
LGTM !

1. **`filterSearch` Addition:**  
   Nice placeholder with `UnsupportedOperationException`—good for tracking unimplemented features.

2. **`listCategories` Adjustment:**  
   Clean implementation of the `listCategories` function. Using `Request.Builder` for URL handling is straightforward and efficient.